### ISSUE TITLE
[BUG] Allow multiple connections from the same IP address should be enabled by default

### ISSUE THREAD

**Issue by StoneMoe** [linuxserver/docker-qbittorrent#240]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

With the default configuration which is ran by the example docker compose file, the "Advance -> libtorrent Section -> Allow multiple connections from the same IP address" is disabled by default, which will prevent incoming connections work as expected, because network of the container is behind NAT in non-host mode.

therefore only one client will be able to connect to our qbittorrent docker instance because they share same NAT address.
in my case, it's 192.168.80.1, which is docker network gateway IP address.

### Expected Behavior

multiple peers should be able to connect to our instance in the same time.

### Steps To Reproduce

1. Setup a instance with compose file in README
2. download something popular
3. there will always only one "I" Flag client show in the peer list
4. enable the multiple connection option I mentioned above
5. there are multiple peers connect to our instance now, share same docker gateway IP but with different port.

### Environment

```markdown
- OS:DSM7
- How docker service was installed: DSM Package Center
```

### CPU architecture

x86-64

### Docker creation

```bash
---
version: "2.1"
services:
  qbittorrent:
    image: lscr.io/linuxserver/qbittorrent:latest
    container_name: qbittorrent
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Etc/UTC
      - WEBUI_PORT=8080
    volumes:
      - /path/to/appdata/config:/config
      - /path/to/downloads:/downloads
    ports:
      - 8080:8080
      - 6881:6881
      - 6881:6881/udp
    restart: unless-stopped
```

### Container logs

```bash
No related log
```

**Comment by aptalca** [comment]
Pretty sure that is qbit's default. We don't set that.

**Comment by sub20hz** [comment]
This ended up saving me so much time. I couldn't figure out why subsequent qbittorrent clients running on the same host would not connect to anything. Could we at least make a note in the documentation that this setting will likely need to be changed if running multiple qbittorrent containers on the same host?

**Comment by j0nnymoe** [comment]
I would recommend asking qbittorrent to document this within their own documentation (unless it's there already). We put enough info in our documentation that is relevant to running the containers and anything the might be slightly different to running it bare metal (WEBUI_PORT for example). 

If we started putting little notes like this for niche use cases, our already long readme's would be a lot longer.

**Comment by j0nnymoe** [comment]
Closing as we believe this is something that should be documented at qbittorrent and not within our container docs.

**Comment by StoneMoe** [comment]
> Pretty sure that is qbit's default. We don't set that.

@aptalca actually, libtorrent set that as default.


> I would recommend asking qbittorrent to document this within their own documentation (unless it's there already). We put enough info in our documentation that is relevant to running the containers and anything the might be slightly different to running it bare metal (WEBUI_PORT for example).

@j0nnymoe since qBittorrent doesn't officially support container usage, I believe they might not be concerned with or prioritize this issue.

It's worth mentioning that although the container is running without crashing, the ability to handle multiple incoming connections is a fundamental feature of a BitTorrent client. Overlooking this malfunction caused by containerization, especially considering it's included in the default compose file in the README, should be addressed in my opinion.

> If we started putting little notes like this for niche use cases, our already long readme's would be a lot longer.

However, it's important to note that handling multiple incoming connections is not a niche requirement but rather a fundamental aspect of a BitTorrent client. Therefore, it doesn't make sense to disregard it.

If your readme is already long, perhaps you could consider adding a table of contentsor dividing it into multiple sections to make it more manageable. Removing redundant descriptions could also help accommodate the addition of useful information instead of outright refusing it.

**Comment by StoneMoe** [comment]
> This ended up saving me so much time. I couldn't figure out why subsequent qbittorrent clients running on the same host would not connect to anything. Could we at least make a note in the documentation that this setting will likely need to be changed if running multiple qbittorrent containers on the same host?

I want to make it clear that this problem is not about running multiple peer clients on the same host.

The issue is with the way libtorrent is set up by default, which doesn't work well with container usage.  
Because of this, the seed client can't handle more than one incoming peer client at the same time.

**Comment by StoneMoe** [comment]
And I kindly request that you refrain from closing issues like this, as it hinders productive conversation and resolution.

**Comment by aptalca** [comment]
To circle back, qbit running in a docker bridge network does not mean it's by default NAT'ed in a way that all incoming connections look like they're coming from the same IP (a docker network IP).

If you're observing that, you likely have other issues with your docker system or the docker network you're using.

The setting you're referring to is not needed to be enabled by most if not all users of this container when run in a supported manner. Upstream project made that setting available for allowing multiple connections from the same remote IP, not to work around a networking issue like the one you're having. They also recommend against enabling it: https://www.libtorrent.org/reference-Settings.html#allow_multiple_connections_per_ip

In short, that setting does not need to be enabled in our container. And what you're observing is not a limitation or a side effect of running qbit/libtorrent in a container.

In my case, with the default settings, qbit in a user defined docker bridge network (vanilla one created by compose automatically), qbit accurately determines the remote IPs and I do get simultaneous incoming connections.

If you're interested in getting the bottom of your issue, I suggest you take a step back and evaluate your docker install and networking. Making broad assumptions and making demands based on those false assumptions are not the way to get assistance. Just my $0.02.

**Comment by StoneMoe** [comment]
I apologize for not using more appropriate language to express my speculation about the issue.  
Thank you for taking the time to provide a detailed explanation of your previous response.

I don't understand what you mean by "not NAT'ed in a way that all incoming connections look like they're coming from the same IP", normally that's how NAT works.
Also I following your advice, I rechecked the environment and configuration, and run the same test on multiple systems, but encountered the same problem, from my perspective, I am almost certain that this is the default network behavior of Docker.

I understand that resolving issues that cannot be easily reproduced can be challenging.  
Nonetheless, appreciate your response.
